### PR TITLE
Added application peformance metrics to influxdb

### PR DESCRIPTION
**Pull request by tech-sam**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -29,3 +30,27 @@ services:
...
+      - INFLUXDB_DB=
+      - INFLUXDB_ADMIN_USER=
+      - INFLUXDB_ADMIN_PASSWORD=  
+  
+  grafana:
+    image: grafana/grafana-oss:latest
+    ports:
+      - 3000:3000
+    volumes:
+      - ./grafana_conf:/var/lib/grafana:rw
```
[the remark this answers is not in the document]
Comment (tech-sam) [reply docker-compose.yaml:54]: agree , added to `.gitignore`

File: pkg/harvester/performance_monitor.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import (
+	"context"
+	"time"
+
+	influx_write "github.com/influxdata/influxdb-client-go/v2/api/write"
+)
+
+type PerformanceMonitorClient interface {
+	// TODO add generics support
+	WriteMetrics(*influx_write.Point, ErrorHandler)
```
Comment (nwasiqUC) [inline pkg/harvester/performance_monitor.go:12]: An interface that takes something specific to influxdb, seems weird

File: pkg/influxdb/client.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	influx_db2 "github.com/influxdata/influxdb-client-go/v2"
+	influx_api "github.com/influxdata/influxdb-client-go/v2/api"
+	"github.com/pkg/errors"
+)
+
+type InfluxDb2Client struct {
+	client influx_api.WriteAPI
+}
+
+func NewPerformanceMonitorClient(cfg *harvester.Config, fn harvester.ErrorHandler) (*InfluxDb2Client, error) {
```
Comment (nwasiqUC) [inline pkg/influxdb/client.go:17]: there's already a "NewPerformanceMonitorClient" function that returns the interface, shouldn't this be named "NewInfluxDBMonitorClient"?

**Review by nwasiqUC — APPROVED** [review]
Left a comment and a question

File: pkg/harvester/performance_monitor.go (hunk 2 above)
Comment (tech-sam) [reply pkg/harvester/performance_monitor.go:12]: this function is not used, and yes TODO added  for generics support reminder

File: pkg/substrate/slashes.go [hunk 4]
```diff
@@ -25,11 +25,19 @@ type Slash struct {
 	Type         string `json:"type"`
 }
 
-func (sh *SubstrateHarvester) ProcessSlashes(ctx context.Context, fn harvester.ErrorHandler) error {
-	return sh.processSlashes(ctx, fn, 5*time.Minute)
+func (sh *SubstrateHarvester) ProcessSlashes(ctx context.Context,
+	fn harvester.ErrorHandler,
+	pmc harvester.PerformanceMonitorClient,
+	topic string) error {
+	return sh.processSlashes(ctx, fn, pmc, topic, 1*time.Second)
```
Comment (rwajon) [inline pkg/substrate/slashes.go:32]: Hey @tech-sam, initially the time interval was 5 minutes(`5*time.Minute`), here it has been changed to 1 second

**Pull request merged**